src/launch/launcher.rs

```diff
@@ -29,6 +29,13 @@ pub struct Launcher<'a, T> {
     sev: Fd<'a>,
 }
 
+impl<'a, T> Launcher<'a, T> {
+    pub fn pin_pages(&mut self, memory: &'a [u8]) -> Result<()> {
+        encrypt_region(&self.kvm, memory)?;
+        Ok(())
+    }
+}
```

Comment:
I'm ambivalent about the placement of this method.

I suspect pinning needs to happen whenever pages are added to the VM. I further suspect this should all happen before we begin the `Launcher` sequence. So I think it should look like this:

1. create a KVM VM
2. add the pages
3. pin all pages (SEV-only)
4. run the `Launcher` sequence (SEV-only)

Perhaps it should be: `Launcher<New>::pin_pages()`?

Reply:
Let me put it this way. If you add pages to the VM after the launch is finished, you'll still need to pin them. So I don't think memory pinning should be connected to the `Launcher` process at all.

Reply:
I agree. I will separate this in my next revision.